pages/owned/index.js

```diff
@@ -11,9 +12,19 @@ export default function Owned() {
   const { data: session, status: sessionStatus } = useSession();
   const [filters, setFilters] = useState({ lightNeed: [], waterNeed: [] });
   const userId = session?.user.id;
-  const { data: plantList, isPlantsLoading } = useSWR(
-    session ? `/api/user/${userId}/owned` : null
-  );
+  const {
+    data: plantList,
+    isPlantsLoading,
+    mutate,
+  } = useSWR(session ? `/api/user/${userId}/owned` : null);
+  const searchParams = useSearchParams();
+  const deleted = searchParams.get("deleted") || undefined;
+  if (deleted) {
+    mutate(
+      plantList.filter((plant) => !(plant._id === deleted)),
+      true
+    );
+  }
```

Comment:
not needed, since useSWR is called automatically when getting sent to this page, no mutate, deleted or similar needed.